gneiss/util.py

```diff
@@ -184,3 +188,81 @@ def rename_internal_nodes(tree, names=None, inplace=False):
             n.name = label
             i += 1
     return _tree
+
+
+def _intersect_of_table_metadata_tree(table, metadata, tree):
+    """ Matches tips, features and samples between the table, metadata
```

Comment:
What about "The intersection of tips and features" and then the more specific detail follows? I believe samples and metadata subset are implicit as driven by the overlap in tips and features, right?